conference/src/main/java/com/kyriba/conference/management/api/ScheduleController.java

```diff
@@ -30,78 +37,74 @@
 @RequestMapping("/api/v1/schedule")
 public class ScheduleController
 {
+  private final ScheduleService scheduleService;
+
+
+  @Autowired
+  public ScheduleController(ScheduleService scheduleService)
+  {
+    this.scheduleService = scheduleService;
+  }
+
 
   @ApiOperation(value = "Show conference schedule")
   @GetMapping(produces = APPLICATION_JSON_UTF8_VALUE)
   ScheduleResponse showSchedule()
   {
-    return ScheduleResponse.builder().presentations(buildPresentations()).build();
+    List<PresentationResponse> presentations = stream(scheduleService.getSchedule().spliterator(), false)
+        .map(this::toPresentationResponse)
+        .collect(Collectors.toList());
+    return ScheduleResponse.builder().presentations(presentations).build();
   }
 
 
   @ApiOperation(value = "Add presentation in conference schedule")
   @PostMapping(value = "/presentations", produces = APPLICATION_JSON_UTF8_VALUE, consumes = APPLICATION_JSON_UTF8_VALUE)
   Long addPresentation(
-      @ApiParam(value = "Presentation create object", required = true) @RequestBody PresentationRequest presentation)
+      @ApiParam(value = "Presentation create object", required = true) @RequestBody PresentationRequest presentationRequest) throws
+      LinkedEntityNotFound
   {
-    return 55L;
+    return scheduleService.addPresentation(presentationRequest);
   }
 
 
   @ApiOperation(value = "View presentation information")
   @GetMapping(value = "/presentations/{id}", produces = APPLICATION_JSON_UTF8_VALUE)
-  PresentationResponse getPresentation(
-      @ApiParam(value = "Presentation identity", required = true) @PathVariable String id)
+  ResponseEntity<PresentationResponse> getPresentation(
+      @ApiParam(value = "Presentation identity", required = true) @PathVariable Long id)
   {
-    return PresentationResponse.builder()
-        .hall(123L)
-        .startTime(LocalTime.of(11, 30))
-        .endTime(LocalTime.of(12, 30))
-        .topic(new TopicDto("Spring Data REST", "Andy Wilkinson"))
-        .build();
+    return ResponseEntity.of(scheduleService.getPresentation(id).map(this::toPresentationResponse));
```

Comment:
Transformation logic should be moved to service

Reply:
done